libcst/metadata/scope_provider.py

```diff
@@ -950,6 +951,30 @@ def _handle_string_annotation(
         self, node: Union[cst.SimpleString, cst.ConcatenatedString]
     ) -> bool:
         """Returns whether it successfully handled the string annotation"""
+        if self.__in___all___stack[-1]:
+            name: str | bytes | None = None
+
+            if isinstance(node, cst.SimpleString):
+                name = node.value
+
+            if isinstance(node, cst.ConcatenatedString):
+                name = node.evaluated_value
+
+            assert isinstance(name, str), f"Expected str, got {type(name)}"
+
+            name = name.replace('"', "")
+
+            access = Access(
+                cst.Name(name),
+                self.scope,
+                is_annotation=False,
+                is_type_hint=False,
+            )
+
+            list(self.scope.assignments[name])[0].record_access(access)
```

Comment:
I would add an entry into `self.__deferred_accesses` instead, to make sure this `Access` gets the same treatment as any other.